src/circuit.rs

```diff
@@ -92,6 +92,91 @@ pub struct Cell {
     column: Column<Any>,
 }
 
+/// An assigned cell.
+#[derive(Clone, Copy, Debug)]
+pub struct AssignedCell<F: Field, T: Copy + Into<F>> {
+    value: Option<T>,
+    cell: Cell,
+    _marker: PhantomData<F>,
+}
+
+impl<F: Field, T: Copy + Into<F>> AssignedCell<F, T> {
+    /// Assign a new AssignedCell.
+    pub fn assign<A, AR>(
+        region: &mut Region<'_, F>,
+        annotation: A,
+        column: Column<Any>,
```

Comment:
Idea: `column: impl Into<Column<Any>>` (maybe we can do this in other places too...)